buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -724,5 +724,27 @@ protected void memoryCopy(ByteBuffer src, int srcOffset, ByteBuffer dst, int dst
                 dst.put(src);
             }
         }
+
+        @Override
+        protected void finalize() throws Throwable {
+            try {
+                super.finalize();
+            } finally {
+                destroyPoolSubPages(smallSubpagePools);
+                destroyPoolSubPages(tinySubpagePools);
+                qInit.destroy(this);
+                q000.destroy(this);
+                q025.destroy(this);
+                q050.destroy(this);
+                q075.destroy(this);
+                q100.destroy(this);
+            }
+        }
+
+        private void destroyPoolSubPages(PoolSubpage<ByteBuffer>[] pages) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)